packages/vite/src/node/utils.ts

```diff
@@ -1045,19 +1049,27 @@ export function resolveServerUrls(
       })
   }
 
-  const cert =
-    httpsOptions?.cert && !Array.isArray(httpsOptions.cert)
-      ? new crypto.X509Certificate(httpsOptions.cert)
-      : undefined
-  const hostnameFromCert = cert?.subjectAltName
-    ? extractHostnamesFromSubjectAltName(cert.subjectAltName)
-    : []
+  const certs = httpsOptions?.cert ? arraify(httpsOptions.cert) : []
+  const hostnameFromCert = certs
+    .map((cert) => {
+      try {
+        const buf = bufferify(cert)
+        return new crypto.X509Certificate(buf)
```

Comment:
```suggestion
        return new crypto.X509Certificate(cert)
```
I think this conversion is handled internally by `X509Certificate`.

Reply:
I added this conversion mainly because of the mock work during testing, but if we extract it into a helper function as suggested, I think it can be removed as you said. Thanks!